ddl/ddl_worker.go

```diff
@@ -367,14 +370,24 @@ func (d *ddl) addBatchDDLJobs2Table(tasks []*limitJobTask) error {
 		return nil
 	})
 	if err == nil {
-		jobTasks := make([]*model.Job, len(tasks))
+		jobTasks := make([]*model.Job, 0, len(tasks))
 		for i, task := range tasks {
 			job := task.job
 			job.Version = currentVersion
 			job.StartTS = startTS
 			job.ID = ids[i]
 			setJobStateToQueueing(job)
-			jobTasks[i] = job
+
+			if d.stateSyncer.IsUpgradingState() && !hasSysDB(job) {
+				if err = pauseRunningJob(sess.NewSession(se), job, model.AdminCommandBySystem); err != nil {
```

Comment:
It would be better to use PauseJobs() instead of pauseRunningJob() here:
- PauseJobs() is an interface to other modules; ddl_worker is different from ddl/ddl.go
- PauseJobs() processes multiple jobs together in one transaction; Normally it would be reduce this loop

Reply:
* They are in the the same pkg。
* This logic needn't a new txn, we will do it in `insertDDLJobs2Table`